Shared/Actions/ContextParams.cs

```diff
@@ -52,9 +58,24 @@ public override void Check()
         {
             SortedDictionary<string, object> dict = base.ToParamsDictionary();
 
-            AddParam(dict, Constants.CONTEXT_PARAM_NAME, Context);
+            var sbContext = new StringBuilder();
+            
+            if (ContextDict != null && ContextDict.Count > 0)
+                sbContext.Append(string.Join("|", ContextDict.SafePairs));
+
+            if (!string.IsNullOrEmpty(Context))
+            {
+                if (sbContext.Length > 0)
```

Comment:
Instead of `StringBuilder` I would just append `Context` to the list of SafePairs:
```
var contextPairs = ContextDict.SafePairs;
```
perform all kinds of checks if null, etc
```
contextPairs.Add(Context);
```
and in the end 
```
string.Join("|", contextPairs)
```